### PR TITLE
Update .NET SDK to v6.0

### LINKED ISSUE #47: Update to .NET 6.0
After Dec 3 2022 .NET 3.1 will no longer be supported or patched.

Related to https://github.com/microsoft/component-detection/issues/107

### PR DESCRIPTION
**Pull request by aasim**
Closes #47

### CODE REVIEW

File: docs/building-from-source.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Building the SBOM tool from source.
 
-The SBOM tool is a cross-platform tool written in C# and is compiled using .NET Core 3.1. Follow the instructions below to build the tool from source.
+The SBOM tool is a cross-platform tool written in C# and is compiled using .NET Core 6. Follow the instructions below to build the tool from source.
```
Comment (ksigmund) [inline docs/building-from-source.md:3]: I think it's just ".NET 6"

File: global.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"sdk": {
-		"version": "3.1.300",
+		"version": "6",
```
Comment (JamieMagee) [inline global.json:3]: If you set a specific version here `actions/setup-dotnet` will read this file, and you can omit `dotnet-version` setting. `6.0.400` is the latest version.

Comment (aasim) [reply global.json]: Fixed

File: docs/building-from-source.md (hunk 1 above)
Comment (aasim) [reply docs/building-from-source.md:3]: Fixed

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -58,7 +58,7 @@ jobs:
     - name: Setup .NET
       uses: actions/setup-dotnet@v2
       with:
-        dotnet-version: 3.1.300
+        dotnet-version: 6.0.400
```
Comment (edgarrs) [inline .github/workflows/codeql-analysis.yml:61]: Do you still need to specify the version here? I see it is not specified other yml files

Comment (aasim) [reply .github/workflows/codeql-analysis.yml]: Missed it, removed

**Review by edgarrs — APPROVED** [review]

**Pull request merged**